packages/runtime/src/components/shared/CompatNextImage/CompatNextImage.ts

```diff
@@ -0,0 +1,11 @@
+import CurrentNextImage from 'next/image'
+
+let NextImage = CurrentNextImage
+
+try {
+  // If on Next.js v13, we're using next/legacy/image
+  // TODO: This will throw a warning on a console: Module not found: Can't resolve 'next/legacy/image'
+  NextImage = require('next/legacy/image')
+} catch {}
+
+export const CompatNextImage = NextImage
```

Comment:
Maybe we could check the Next.js version instead of using try/catch?

Reply:
I tried checking the Next.js version and dynamic import using this code:
```ts
export const CompatNextImage = dynamic(() =>
  nextMajorVersion >= 13 ? import('next/legacy/image') : import('next/image'),
) as typeof NextImage
```
But it still gives this error on compile time:
```
wait  - compiling...
error - .yalc/@makeswift/runtime/dist/CompatNextImage.es.js:5:113
Module not found: Can't resolve 'next/legacy/image'
```
Like our problem with the dynamic import for the new pages sidebar, I think Next.js is resolving the import during compile time, so we still can't use this approach.

I'm not sure how to solve this other than using the `require` with a `try catch`. I tried to search on how to make the `import()` function not throw an error, but I couldn't find anything yet.

Reply:
The difference is that we can put `require` in a `try catch`, and basically we catch the error thrown when we try to import `next/legacy/image`.

From what I'm seeing, the Next.js will resolve the `import` first, before even building/compiling the code. 

Yes, I'm also thinking of _adding a layer of indirection with an extra module_. Let me try it.

Reply:
But why would require work any differently than import? At the end of the day webpack needs to create the chunks and it will use the import specifier or module ID to do that. When Next.js builds the browser bundle I would imagine the require module ID is also resolved at build time. Or maybe that's not the case?

Reply:
I'm not sure what's happening. I just can't catch the error if we're using `import`, but it's working if we're using `require`.

Reply:
Well that's interesting. I guess we could look at the build output if we want to see what webpack's doing

Reply:
Hmm, ok. We should be careful using `require`, then. Can you take analyze the bundle using `@next/bundle-analyzer` and make sure that using `require` doesn't result in a problem like a bunch of JavaScript being included that wasn't before?

Also, we should verify that Rollup (via Vite) isn't what's attempting to resolve the `require` at build time when building `@makeswift/runtime`

Reply:
>Also, we should verify that Rollup (via Vite) isn't what's attempting to resolve the require at build time when building @makeswift/runtime

I don't think this is from the Rollup. This is the output from the Rollup:
```js
import dynamic from "next/dynamic";
const nextVersion = require("next/package.json").version;
const nextMajorVersion = parseInt(nextVersion.split(".")[0]);
const CompatNextImage = dynamic(() => nextMajorVersion >= 13 ? import("next/legacy/image") : import("next/image"));
export { CompatNextImage as C };
//# sourceMappingURL=CompatNextImage.es.js.map
```

I tested using `yarn build` on both Next.js v12 and Next.js v13 using the `@next/bundle-analyzer`, and there was no increase in size or `First Load JS`.

But, in Next.js v12, there were these warnings during the build step:
```
./.yalc/@makeswift/runtime/dist/CompatNextImage.es.js
Module not found: Can't resolve 'next/legacy/image' in '/Users/fikrikarim/Desktop/workspace/custom-host-sandbox/.yalc/@makeswift/runtime/dist'

Import trace for requested module:
./.yalc/@makeswift/runtime/dist/CompatNextImage.es.js
./.yalc/@makeswift/runtime/dist/Root.es.js
./.yalc/@makeswift/runtime/dist/index.es.js
./.yalc/@makeswift/runtime/dist/next.es.js
```

Sigh..

[Relevant issue from the Next.js repo](https://github.com/vercel/next.js/issues/18819)